state/cloudcredentials_test.go

```diff
@@ -51,6 +51,38 @@ func (s *CloudCredentialsSuite) TestUpdateCloudCredentialNew(c *gc.C) {
 	c.Assert(out, jc.DeepEquals, expected)
 }
 
+func (s *CloudCredentialsSuite) TesCreateInvalidCredential(c *gc.C) {
+	err := s.State.AddCloud(cloud.Cloud{
+		Name:      "stratus",
+		Type:      "low",
+		AuthTypes: cloud.AuthTypes{cloud.AccessKeyAuthType, cloud.UserPassAuthType},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	cred := cloud.NewCredential(cloud.AccessKeyAuthType, map[string]string{
+		"foo": "foo val",
+		"bar": "bar val",
+	})
+	// Setting of these properties should have no effect when creating a new credential.
+	cred.Invalid = true
```

Comment:
If these values are not supposed to be set when creating a new credential, it should error if they are set

Reply:
Good idea - I'll do that.